gcc/rust/analysis/rust-type-resolution.cc

```diff
@@ -670,11 +675,16 @@ TypeResolution::visit (AST::Function &function)
     }
 
   // walk the expression body
+  localsPerBlock.Push ();
   for (auto &stmt : function.function_body->statements)
     {
       stmt->accept_vis (*this);
     }
 
+  auto localMap = localsPerBlock.Pop ();
```

Comment:
This seems to be a C++17 thing and gcc is warning about it. Not sure if we should allow it or now. Might be something to create an issue out of for what C++ version will we limit ourselves to.